### PR TITLE
[docs] docs: update MoE CUDA graph guidance

### PR DESCRIPTION
**Pull request by yaoyu-33**
## Summary
- Add measured MoE CUDA graph guidance for a Qwen3 30B A3B H100 BF16 short run.
- Clarify that TE-scoped graph capture is valid but not a guaranteed speedup.
- Update the CUDA graph skill CLI example to use the current performance harness and comma-separated scopes.

## Experiment
- Ran eager vs TE-scoped `attn,moe_router,moe_preprocess` CUDA graphs on Qwen3 30B A3B pretrain.
- Scale: 2 H100 nodes, 16 GPUs total, all-to-all dispatcher fixed across both runs.
- Result: graph capture succeeded, but replay iterations were slightly slower than eager on this all-to-all fallback shape.

## Validation
- `UV_CACHE_DIR=.local/uv-cache uv run pre-commit run --all-files`
- No local unit tests run, per handoff.

### CODE REVIEW

**Comment by yaoyu-33** [comment]
Experiment handoff results for the short MoE CUDA graphs comparison.

Context:
- Branch base: latest `origin/main`, not stacked on PR #3867
- Workload: Qwen3 30B A3B pretrain, H100 BF16, config variant v1
- Scale: 2 nodes, 8 GPUs/node, 16 GPUs total, EP=16
- Dispatcher/layout fixed: all-to-all dispatcher, TP=1, PP=1, CP=1, MBS=1, GBS=1024
- Max steps: 8
- Common overrides: `--moe_flex_dispatcher_backend None`, `--moe_a2a_overlap false`, `--tokenizer_type NullTokenizer`, `comm_overlap.overlap_moe_expert_parallel_comm=false`, `comm_overlap.delay_wgrad_compute=false`, `model.moe_shared_expert_overlap=false`, `model.moe_permute_fusion=false`
- Local unit tests: not run, per handoff
- Pre-commit: `UV_CACHE_DIR=.local/uv-cache uv run pre-commit run --all-files` passed

Result table, using replay-window iterations 5-8:

| Case | CUDA graph config | Job | Status | Replay mean step | Replay median step | Replay mean TFLOP/s/GPU | Final loss | Rank-0 peak max allocated | Train log |
|---|---|---:|---|---:|---:|---:|---:|---:|---|
| `A_eager` | `cuda_graph_impl=none`, empty scope | `11855626` | completed | 41.36 s | 41.36 s | 145.8 | 10.34096 | 61.351 GB | `/lustre/fsw/IOR-test/yuya/moe_cuda_graphs_20260518/logs/A_eager_11855626_train.log` |
| `B_te_attn_router_preprocess` | `transformer_engine`, `attn,moe_router,moe_preprocess` | `11855627` | completed | 42.00 s | 41.98 s | 143.6 | 10.31410 | 61.351 GB before capture log point | `/lustre/fsw/IOR-test/yuya/moe_cuda_graphs_20260518/logs/B_te_attn_router_preprocess_11855627_train.log` |

Additional artifacts:
- Slurm log, eager: `/lustre/fsw/IOR-test/yuya/moe_cuda_graphs_20260518/logs/moe-cg_11855626.out`
- Slurm log, TE-scoped: `/lustre/fsw/IOR-test/yuya/moe_cuda_graphs_20260518/logs/moe-cg_11855627.out`
- Eager config: `/lustre/fsw/IOR-test/yuya/moe_cuda_graphs_20260518/configs/A_eager_11855626.yaml`
- TE-scoped config: `/lustre/fsw/IOR-test/yuya/moe_cuda_graphs_20260518/configs/B_te_attn_router_preprocess_11855627.yaml`
- Local report: `.local/moe_cuda_graphs_experiment_20260518.md`

Graph evidence:
- `INFO:megatron.core.transformer.cuda_graphs:Rank 0: 48 graphable layers.`
- `INFO:megatron.core.transformer.cuda_graphs:Start CUDA Graphs capture...`
- `INFO:megatron.core.transformer.cuda_graphs:Time spent in CUDA Graphs capture on rank 0: 6.906740665435791s`
- Shutdown cleanup: `Rank 0: 0 graphs deleted with explicit reset, 48 graphs deleted without explicit reset.`

Per-iteration data:

| Case | Iter 1 | Iter 2 | Iter 3 | Iter 4 | Iter 5 | Iter 6 | Iter 7 | Iter 8 |
|---|---:|---:|---:|---:|---:|---:|---:|---:|
| Eager step time | 87.883 s | 44.930 s | 41.375 s | 41.392 s | 41.396 s | 41.366 s | 41.338 s | 41.345 s |
| TE-scoped step time | 88.427 s | 44.814 s | 41.469 s | 54.549 s | 42.132 s | 41.928 s | 41.999 s | 41.960 s |

Effective config checks:
- Eager: `cuda_graph_impl: none`, `cuda_graph_scope: []`, `use_te_rng_tracker: false`, `rng.te_rng_tracker: false`
- TE-scoped: `cuda_graph_impl: transformer_engine`, `cuda_graph_scope: [attn, moe_router, moe_preprocess]`, `cuda_graph_warmup_steps: 3`, `use_te_rng_tracker: true`, `rng.te_rng_tracker: true`
- Both: `moe_token_dispatcher_type: alltoall`, `moe_permute_fusion: false`

Caveats:
- This is a short graph bring-up and replay-window comparison, not a full production benchmark.
- The all-to-all dispatcher was fixed intentionally because the previous container did not include DeepEP/HybridEP packages; graph behavior may differ with a working flex backend.
- `model.moe_permute_fusion=false` was applied to both runs because the same PyTorch 25.11 / TE / Triton container failed fused MoE permutation in the previous dispatcher smoke.
- The two jobs ran concurrently on different nodes, which keeps the comparison compact but does not eliminate cluster variance.
- Rank-0 memory logging did not emit a post-capture memory line, so the memory value is the available peak allocation line rather than a precise graph memory delta.
- Losses were close and trended together, but not bitwise identical.

Recommendation: TE-scoped `attn,moe_router,moe_preprocess` graphs are valid for this Qwen3 MoE path and capture cleanly, but they did not improve replay timing on this H100 all-to-all fallback shape. Validate replay iterations on the target dispatcher/container instead of treating successful capture as an automatic throughput win.

**Review by cuichenx — APPROVED** [review]
